src/ucs/datastruct/conn_match.c

```diff
@@ -157,19 +159,35 @@ ucs_conn_sn_t ucs_conn_match_get_next_sn(ucs_conn_match_ctx_t *conn_match_ctx,
 {
     ucs_conn_match_peer_t *peer = ucs_conn_match_get_conn(conn_match_ctx,
                                                           address);
-    return peer->next_conn_sn++;
+    ucs_conn_sn_t conn_sn       =
+            (peer->next_conn_sn != conn_match_ctx->max_conn_sn) ?
```

Comment:
maybe this function should fail if reached conn_match_ctx->max_conn_sn?

Reply:
why? we have to remove maximum conn_sn and it disables connection matching when the maximum conn_sn value will be passed to conn_match_insert and etc

Reply:
it looks weird to have check for max_conn_sn here and not do ++
maybe not call this function at all if reached max sn?

Reply:
but then user have to check whether it reached max sn or not
why not check it here?